nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py

```diff
@@ -226,24 +228,60 @@ def __init__(
         folder_path: FolderPathType,
         keep_mua_units: bool = True,
         exclude_shanks: Optional[list] = None,
-        load_waveforms: bool = False,
-        gain: Optional[float] = None,
+        xml_file_path: OptionalFilePathType = None,
+        # TODO: we can enable this once
+        #     a) waveforms on unit columns support conversion factor in NWB
+        #     b) write_sorting utils support writing said waveforms properly to a units table
+        # load_waveforms: bool = False,
+        # gain: Optional[float] = None,
     ):
+        """
+        Load and prepare spike sorted data and corresponding metadata from the Neuroscope format (.res/.clu files).
+
+        Parameters
+        ----------
+        folder_path : FolderPathType
+            Path to folder containing .clu and .res files.
+            The default is None.
+        keep_mua_units : bool
+            Optional. Whether or not to return sorted spikes from multi-unit activity.
+            The default is True.
+        exclude_shanks : list
+            Optional. List of indices to ignore. The set of all possible indices is chosen by default, extracted as the
+            final integer of all the .res.%i and .clu.%i pairs.
+        xml_file_path : OptionalFilePathType, optional
+            Path to .xml file containing device and electrode configuration.
+            If unspecified, it will be automatically set as the only .xml file in the same folder as the .dat file.
+            The default is None.
+        load_waveforms : bool, optional
+            If True, extracts waveform data from .spk.%i files in the path corresponding to
+            the .res.%i and .clue.%i files and sets these as unit spike features.
+            The default is False.
+            Not currently in use pending updates to NWB waveforms.
+        gain : float, optional
+            If loading waveforms, this value converts the data type of the waveforms to units of microvolts.
+            Conversion factors from int16 to Volts are not contained in xml_file_path; set them explicitly here.
+            Most common value is 0.195 for an intan recording system.
+            The default is None.
+            Not currently in use pending updates to NWB waveforms.
+        """
         assert HAVE_LXML, INSTALL_MESSAGE
 
         super().__init__(
             folder_path=folder_path,
             keep_mua_units=keep_mua_units,
             exclude_shanks=exclude_shanks,
-            load_waveforms=load_waveforms,
-            gain=gain,
+            xml_file_path=xml_file_path,
```

Comment:
This generates an error as it stands: the class from where this inherited does not expect a `xml_file_path`.

    ---------------------------------------------------------------------------
    TypeError                                 Traceback (most recent call last)
    /tmp/ipykernel_11701/3205164773.py in <module>
         78         print(clu_matches_in_session[0].name, clu_matches_in_session[0].suffix)
         79         if '1' in clu_matches_in_session[0].suffix:
    ---> 80             multisorting_neuroscope = NeuroscopeSortingInterface(folder_path=session_path, keep_mua_units=False, xml_file_path=str(xml_file_path))
         81             source_data.update(
         82                 NeuroscopeSorting=dict(
    
    ~/development/nwb-conversion-tools/nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py in __init__(self, folder_path, keep_mua_units, exclude_shanks, xml_file_path)
        272             keep_mua_units=keep_mua_units,
        273             exclude_shanks=exclude_shanks,
    --> 274             xml_file_path=xml_file_path,
        275             # TODO: we can enable this once
        276             #     a) waveforms on unit columns support conversion factor in NWB
    
    ~/development/nwb-conversion-tools/nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py in __init__(self, **source_data)
         26     def __init__(self, **source_data):
         27         super().__init__(**source_data)
    ---> 28         self.sorting_extractor = self.SX(**source_data)
         29         self.writer_class = map_si_object_to_writer(self.sorting_extractor)(self.sorting_extractor)
         30 
    
    TypeError: __init__() got an unexpected keyword argument 'xml_file_path'

Reply:
@h-mayorquin You will have to make sure you are also using `master` of `spikeextractors` until they do a new release, see https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py#L576